src/rt/ds/hashmap.h

```diff
@@ -39,7 +49,32 @@ namespace verona::rt
     {
       // The Object* returned will retain its RememberedSet mark bit, but not
       // its dib.
-      return (Object*)(p & ~(size_t)DIB_MAX);
+      return (Object*)(p & ~DIB_MAX);
+    }
+
+    template<typename>
```

Comment:
I wonder if this would be clearer as something like
```C++
    template<typename>
    struct decompose_entry_type : std::false_type
    {
    };

    template<>
    struct decompose_entry_type<Object*> : std::true_type
    {
      using value_type = Object*;
      static constexpr bool is_set = true;
    };

    template<typename V>
    struct decompose_entry_type<std::pair<Object*, V>> : std::true_type
    {
      using value_type = V;
      static constexpr bool is_set = false;
    };
```
At the moment, `is_valid_pair` is performing two jobs, but that isn't clear on first reading.

```C++
    static_assert(
      decompose_entry_type<Entry>::value,
      "Map Entry must be Object* or std::pair<Object*, V>");

    using ValueType = typename decompose_entry_type<Entry>::value_type;
    static constexpr bool is_set = decompose_entry_type<Entry>::is_set;
```
```